openmetadata-ui/src/main/resources/ui/src/components/MyData/MyDataWidget/MyDataWidget.test.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright 2023 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { act, render, screen } from '@testing-library/react';
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { getUserById } from 'rest/userAPI';
+import { MyDataWidget } from './MyDataWidget.component';
+
+const userDetails = {
+  id: '123',
+};
+
+jest.mock('rest/userAPI', () => ({
+  getUserById: jest.fn().mockImplementation(() =>
+    Promise.resolve([
+      {
+        id: '1',
+        name: 'test 1',
+        fullyQualifiedName: 'test-1',
+      },
+      {
+        id: '2',
+        name: 'test 2',
+        fullyQualifiedName: 'test-2',
+      },
+    ])
+  ),
+}));
+
+jest.mock('utils/EntityUtils', () => ({
+  getEntityName: jest.fn().mockImplementation((obj) => obj.name),
+}));
+
+jest.mock('utils/TableUtils', () => ({
+  getEntityLink: jest.fn().mockImplementation((link) => link),
+  getEntityIcon: jest.fn().mockImplementation((obj) => obj.name),
+}));
+
+jest.mock('./../../../AppState', () => ({
+  getCurrentUserDetails: jest.fn().mockImplementation(() => {
+    return userDetails;
+  }),
+}));
+
+jest.mock(
+  'components/Skeleton/MyData/EntityListSkeleton/EntityListSkeleton.component',
+  () => {
+    return jest.fn().mockImplementation(({ children }) => <>{children}</>);
+  }
+);
+
+describe('MyDataWidget component', () => {
+  it('should fetch data', () => {
+    act(() => {
+      render(<MyDataWidget />);
+    });
+
+    expect(getUserById).toHaveBeenCalledWith('123', 'owns');
+  });
+
+  it('should render header', () => {
+    act(() => {
+      render(
+        <MemoryRouter>
+          <MyDataWidget />
+        </MemoryRouter>
+      );
+    });
+
+    expect(screen.getByText('label.my-data')).toBeInTheDocument();
+  });
+
+  it('should not render view all for 0 length data', () => {
+    act(() => {
+      render(
+        <MemoryRouter>
+          <MyDataWidget />
+        </MemoryRouter>
+      );
+    });
+
+    expect(screen.queryByTestId('my-data')).not.toBeInTheDocument();
```

Comment:
Can we update the test id? something like `my-data-view-all`

Reply:
Why?

Reply:
Got it udpated